types/exorcist/index.d.ts

```diff
@@ -1,21 +1,88 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="through" />
+/// <reference types="node" />
 
-declare module 'exorcist' {
-    import through = require("through");
+import { Writable } from "stream";
+import { ThroughStream } from "through";
 
-    /** Externalizes the source map found inside a stream to an external .map file.
-     * Works with both JavaScript and CSS input streams
-     * @param file full path to the map file to which to write the extracted source map
-     * @param [url] full URL to the map file, set as sourceMappingURL in the streaming output (default: file)
-     * @param [root] root URL for loading relative source paths, set as sourceRoot in the source map (default: "")
-     * @param [base] base path for calculating relative source paths (default: use absolute paths)
-     * @param [errorOnMissing] when truthy, causes 'error' to be emitted instead of 'missing-map' if no map was found in the stream (default: falsey)
-     */
-    function exorcist(file: string, url?: string, root?: string, base?: string, errorOnMissing?: boolean): through.ThroughStream;
-    export = exorcist;
+export = exorcist;
+
+/**
+ * Externalizes the source map of the file streamed in.
+ *
+ * The source map is written as JSON to `input`, and the original file is streamed out with its
+ * `sourceMappingURL` set to the path of `input` (or to the value of `url`).
+ *
+ * @param input file path or writable stream where the source map will be written
+ * @param url full URL to the map file, set as `sourceMappingURL` in the streaming output (default: `input`)
+ * @param root root URL for loading relative source paths, set as `sourceRoot` in the source map (default: "")
+ * @param base base path for calculating relative source paths (default: use absolute paths)
+ * @param errorOnMissing when truthy, causes `"error"` to be emitted instead of `"missing-map"` if no map was found in the stream (default: falsy)
+ * @returns transform stream into which to pipe the code containing the source map
+ *
+ * @example
+ * import browserify = require("browserify");
+ * import * as path from "path";
+ * import * as fs from "fs";
+ * import exorcist = require("exorcist");
+ *
+ * const mapfile = path.join(__dirname, "bundle.js.map");
+ *
+ * // from a file, to a file, and send source map to its own file
+ * browserify({ debug: true })
+ *     .require(require.resolve("./main"), { entry: true })
+ *     .bundle()
+ *     .pipe(exorcist(mapfile))
+ *     .pipe(fs.createWriteStream(path.join(__dirname, "bundle.js"), "utf8"));
+ *
+ * // from a stream, to a stream, and send source map to a stream
+ * browserify([readableSourceStream], browserifyOptions)
+ *     .bundle()
+ *     .pipe(exorcist(targetSourceMapStream, "/url/path/to/replace/source/comment/with/bundle.js"))
+ *     .pipe(writableTargetStream);
+ */
+declare function exorcist(
+    input: string,
+    url?: string,
```

Comment:
It would be nice for compatibility with the previous version of the definition to make these option parameter types also "| null" so that callers could pass nullable variables to the optional parameters, since this change includes turning on 'strictNullChecks'.
I looked at the the exorcist source code and the usage 'url', 'root', 'base', and 'errorOnMissing' all support both null and undefined (i.e. I don't see any '===/!==' comparisons to 'undefined', 'void 0', or 'typeof undefined' equivalent.

Reply:
Technically, you're right, so done.

Personally, I think it's not very wise to allow `null` values for optional parameters, it's non-idiomatic and very implementation dependant and should the library switch to ES6 optionals it will become plain wrong. And the argument with the backwards compatibility is also not really relevant here because it's a major version bump.

Reply:
I see, understood, I work on projects where strict null checks are partially implemented but null and undefined are interchangeable so these types of definitions cause us issues, but I understand those projects are not in the majority. Thanks for the info.